community/detectors/apache_airflow_exposed_ui/src/main/java/com/google/tsunami/plugins/exposedui/ExposedAirflowServerDetector.java

```diff
@@ -0,0 +1,232 @@
+@ForWebService
+public final class ExposedAirflowServerDetector implements VulnDetector {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+  private final Clock utcClock;
+  private final HttpClient httpClient;
+  private final PayloadGenerator payloadGenerator;
+
+  @Inject
+  ExposedAirflowServerDetector(
+      @UtcClock Clock utcClock, HttpClient httpClient, PayloadGenerator payloadGenerator) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient).modify().setFollowRedirects(true).build();
+    this.payloadGenerator = checkNotNull(payloadGenerator);
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+
+    Builder detectionReport = DetectionReportList.newBuilder();
+    matchedServices.stream()
+        .filter(NetworkServiceUtils::isWebService)
+        .filter(this::isMlFlowWebService)
+        .forEach(
+            networkService -> {
+              if (isServiceVulnerableCheckOutOfBandCallback(networkService)) {
+                detectionReport.addDetectionReports(
+                    buildDetectionReport(
+                        targetInfo,
+                        networkService,
+                        "Apache Airflow Server is misconfigured and can be accessed "
+                            + "publicly, Tsunami security scanner confirmed this by sending an HTTP "
+                            + "request with test connection API and receiving the corresponding "
+                            + "callback on tsunami callback server",
+                        Severity.CRITICAL));
+              } else if (isServiceVulnerableCheckResponse(networkService)) {
+                detectionReport.addDetectionReports(
+                    buildDetectionReport(
+                        targetInfo,
+                        networkService,
+                        "Apache Airflow Server is misconfigured and can be accessed "
+                            + "publicly, We confirmed this by checking API endpoint and matching "
+                            + "the responses with our pattern",
+                        Severity.HIGH));
+              }
+            });
+    return detectionReport.build();
+  }
+
+  public boolean isMlFlowWebService(NetworkService networkService) {
+    logger.atInfo().log("probing apache airflow login page - custom fingerprint phase");
+
+    var uriAuthority = NetworkEndpointUtils.toUriAuthority(networkService.getNetworkEndpoint());
+    var loginPageUrl = String.format("http://%s/%s", uriAuthority, "login");
+    try {
+      HttpResponse loginResponse =
+          this.httpClient.send(get(loginPageUrl).withEmptyHeaders().build());
+
+      if (loginResponse.status() == HttpStatus.OK
+          && loginResponse.bodyString().isPresent()
+          && loginResponse.bodyString().get().contains("Sign In - Airflow")) {
```

Comment:
This could be misleading, since it only checks if the `Sign In - Airflow` is present in the body of the response. You could use a small HTML parsers to find out if the aforementioned string is in the `<title>` tag. 
Moreover i'm noticing that in the login page there are several strings that can be used to perform the fingerprint. For example there is `<a href="https://airflow.apache.org" data-original-title="" title="">Airflow Website</a>`.